lib/libzfs/libzfs_dataset.c

```diff
@@ -196,6 +196,16 @@ zfs_validate_name(libzfs_handle_t *hdl, const char *path, int type,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "reserved disk name"));
 				break;
+			
+			case NAME_ERR_SELF_REFERENCE:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "self reference, ('.') is found in name"));
```

Comment:
I don't think we need two enclosures for the reference, can we do without the parenthesis?

Reply:
Done!